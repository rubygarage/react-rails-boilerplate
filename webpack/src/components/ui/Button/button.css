.btn {
  display: inline-block;
  font-size: 1.6rem;
  line-height: 1.5;
  padding: 0.8rem 2rem;
  background: none;
  text-align: center;
  cursor: pointer;
  transition: 0.15s all ease-in-out;
  border-radius: 0.3rem;
  border-style: solid;
  border-width: 0.2rem;
  border-color: transparent;
  position: relative;

  &[disabled] {
    opacity: 0.8;
    color: rgba(255, 255, 255, 0.3);
    background-color: rgba(109,109,109,0.3);

    &:hover {
      background-color: rgba(109,109,109,0.7);
      cursor: not-allowed;
    }
  }
}

.xs {
  font-size: 1.3rem;
  font-weight: 500;
  line-height: 1.4;
  padding: 0.5rem 1.5rem;
  border-width: 0.1rem;
}

.sm {
  font-size: 1.4rem;
  font-weight: 500;
  line-height: 1.5;
  padding-top: 0.6rem;
  padding-bottom: 0.6rem;
  border-width: 0.1rem;
}

.md {
  line-height: 1.25;
  min-width: 9rem;
  text-transform: uppercase;
  font-weight: bold;
}

.lg {
  font-size: 1.8rem;
  padding-top: 0.9rem;
  padding-bottom: 1rem;
}

.blue-dark {
  color: white;
  background-color: #3c599b;

  &:hover {
    background-color: rgba(60, 89, 155, 0.7);
  }
}

.blue-light {
  color: white;
  background-color: #32cdfd;

  &:hover {
    background-color: rgba(50, 205, 253, 0.7);
  }
}

.blue,
.blue-grey {
  color: white;
  background-color: #627fa0;

  &:hover {
    background-color: rgba(98, 127, 160, 0.7);
  }
}

.grey {
  color: white;
  background-color: #818181;

  &:hover {
    background-color: rgba(129, 129, 129, 0.7);
  }
}

.red {
  color: white;
  background-color: #e94f3b;

  &:hover {
    background-color: rgba(233, 79, 59, 0.7);
  }
}

.green {
  color: white;
  background-color: #47cc52;

  &:hover {
    background-color: rgba(71, 204, 82, 0.7);
  }
}

.orange {
  color: white;
  background-color: #f2a200;

  &:hover {
    background-color: rgba(242, 162, 0, 0.7);
  }
}

.pink {
  color: white;
  background-color: #ff4d4d;

  &:hover {
    background-color: rgba(255, 77, 77, 0.7);
  }
}

.magenta {
  color: white;
  background-color: #c6f;

  &:hover {
    background-color: rgba(204, 102, 255, 0.7);
  }
}

.aquamarine {
  color: white;
  background-color: #3c9;

  &:hover {
    background-color: rgba(51, 204, 153, 0.7);
  }
}

.white {
  color: #4a4a4a;
  background-color: white;

  &:hover {
    opacity: 0.6;
  }
}

.clear-white {
  color: white;
  border-color: white;

  &:hover {
    opacity: 0.6;
  }
}

.clear-grey {
  color: #6d6d6d;
  border-color: #6d6d6d;

  &:hover {
    opacity: 0.6;
  }
}

.clear-blue {
  color: #627fa0;
  border-color: #627fa0;

  &:hover {
    opacity: 0.6;
  }
}

.clear-black {
  color: black;
  border-color: black;

  &:hover {
    opacity: 0.6;
  }
}
