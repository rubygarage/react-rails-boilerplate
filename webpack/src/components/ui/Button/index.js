import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

import styles from './button.css';

function Button({
  children, href, to, color, size, className, disabled, onClick, type,
}) {
  const classes = classNames(styles.btn, styles[color || 'blue'], styles[size], className);

  if (href) {
    return (
      <a
        href={href}
        target="_blank"
        rel="noopener noreferrer"
        className={classes}
        disabled={disabled}
        onClick={onClick}
      >
        {children}
      </a>
    );
  }

  if (to) {
    return (
      // eslint-disable-next-line
      <Link to={to} className={classes} disabled={disabled} onClick={onClick}>{children}</Link>
    );
  }

  return (
    <button type={type} className={classes} disabled={disabled} onClick={onClick}>{children}
    </button>
  );
}

Button.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.string,
  disabled: PropTypes.bool,
  children: PropTypes.node,
  href: PropTypes.string,
  onClick: PropTypes.func,
  to: PropTypes.string,
  type: PropTypes.string,
};

Button.defaultProps = {
  href: '',
  to: '',
  size: 'md',
};

export default Button;
