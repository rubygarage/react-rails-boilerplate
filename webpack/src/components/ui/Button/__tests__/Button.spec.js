import React from 'react'
import { shallow } from 'enzyme'
import Button from 'components/ui/Button'

describe('<Button />', () => {
  let props = {
    className: 'grey',
    disabled: true,
    onClick: jest.fn()
  }
  it('renders Button', () => {
    const component = shallow(<Button />)

    expect(component).toMatchSnapshot()
  })

  it('renders A', () => {
    const component = shallow(<Button href="/" />)

    expect(component).toMatchSnapshot()
  })

  it('renders Link', () => {
    const component = shallow(<Button to="/" />)

    expect(component).toMatchSnapshot()
  })

  it('renders Button with props', () => {
    props = {
      ...props,
      type: 'button'
    }

    const component = shallow(<Button {...props} />)

    expect(component).toMatchSnapshot()
  })

  it('renders A with props', () => {
    props = {
      ...props,
      type: 'button',
      href: '/'
    }

    const component = shallow(<Button {...props} />)

    expect(component).toMatchSnapshot()
  })

  it('renders Link with props', () => {
    props = {
      ...props,
      type: 'button',
      to: '/'
    }

    const component = shallow(<Button {...props} />)

    expect(component).toMatchSnapshot()
  })
})
