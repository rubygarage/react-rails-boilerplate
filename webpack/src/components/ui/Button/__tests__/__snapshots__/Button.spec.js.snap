// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders A 1`] = `
<a
  className="btn blue md"
  href="/"
  rel="noopener noreferrer"
  target="_blank"
/>
`;

exports[`<Button /> renders A with props 1`] = `
<a
  className="btn blue md grey"
  disabled={true}
  href="/"
  onClick={[Function]}
  rel="noopener noreferrer"
  target="_blank"
/>
`;

exports[`<Button /> renders Button 1`] = `
<button
  className="btn blue md"
/>
`;

exports[`<Button /> renders Button with props 1`] = `
<button
  className="btn blue md grey"
  disabled={true}
  onClick={[Function]}
  type="button"
/>
`;

exports[`<Button /> renders Link 1`] = `
<Link
  className="btn blue md"
  replace={false}
  to="/"
/>
`;

exports[`<Button /> renders Link with props 1`] = `
<a
  className="btn blue md grey"
  disabled={true}
  href="/"
  onClick={[Function]}
  rel="noopener noreferrer"
  target="_blank"
/>
`;
